### PR TITLE
updates on movie.salesCast section

### PR DESCRIPTION
**Pull request by phpgeekfr**
issue  #1603

### CODE REVIEW

File: libs/movie/src/lib/movie/form/sales-cast/sales-cast.form.ts [hunk 1]
```diff
@@ -35,18 +37,52 @@ export class MovieSalesCastForm extends FormEntity<MovieSalesCastControl>{
...
+    return this.get('producers');
   }
 
-  public removeCredit(i: number): void {
-    this.credits.removeAt(i);
+  get crew() {
+    return this.get('crew');
+  }
+
+  public addCredit(credit?: Partial<Credit>, type: 'cast' | 'crew' | 'producer' = 'cast'): void {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/sales-cast/sales-cast.form.ts]: I'm pretty sure those functions won't be used as we are going to pass the FormList directly (for next time ;) )

Comment (phpgeekfr) [reply libs/movie/src/lib/movie/form/sales-cast/sales-cast.form.ts]: Yeah I know, update was made to pass the build and to keep a working app

File: libs/utils/src/lib/common-interfaces/identity.ts [hunk 2]
```diff
@@ -103,3 +123,15 @@ export function createCredit(params: Partial<Credit> = {}): Credit {
     ...params
   };
 }
+
+export function createProducer(params: Partial<Credit> = {}): Producer {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/common-interfaces/identity.ts]: I don't really see the point of those factories. If you want to cast to type from Credit to Proucer you could do : 
```typesrcipt
export function createCredit<T extends Credit>(params: Partial<T> = {}): T
```

Comment (phpgeekfr) [reply libs/utils/src/lib/common-interfaces/identity.ts]: I'm not familiar with this `T` things, It's not automatic for me :) Do you want me to change this or you will do it yourself ?

**Pull request merged**